Yosemite/YosemiteTests/Stores/ReceiptStoreTests.swift

```diff
@@ -515,6 +515,62 @@ final class ReceiptStoreTests: XCTestCase {
         let expectedError = expectedResult.failure
         XCTAssertNotNil(expectedError)
     }
+
+    func test_sendReceipt_when_order_updates_and_action_succeeds() throws {
+        let email = "test@test.com"
+        let orderID: Int64 = 56
+        let siteID: Int64 = 123
+        let mockOrder = Order.fake().copy(siteID: siteID, orderID: orderID)
+        let receiptStore = ReceiptStore(dispatcher: dispatcher,
+                                        storageManager: storageManager,
+                                        network: network,
+                                        receiptPrinterService: receiptPrinterService,
+                                        fileStorage: MockInMemoryStorage())
+        network.simulateResponse(requestUrlSuffix: "orders/\(orderID)/actions/send_order_details", filename: "orders-actions-send-order-details")
+        let mockProcessor = MockActionsProcessor()
+        dispatcher.register(processor: mockProcessor, for: OrderAction.self)
+
+        let expectedResult: Result<Yosemite.Order, Error> = waitFor { promise in
+            let action = ReceiptAction.sendReceipt(order: mockOrder, email: email) { result in
+                promise(result)
+            }
+            receiptStore.onAction(action)
+            if case let .updateOrder(_, order, _, _, onCompletion) = mockProcessor.receivedActions.first as? OrderAction {
+                onCompletion(.success(order))
+            }
+        }
+
+        let result = try expectedResult.get()
+        XCTAssert(result.billingAddress?.email == email)
+    }
+
+    func test_sendReceipt_when_order_update_fails_then_send_receipt_fails() throws {
```

Comment:
super nit: this test case might not need to throw?
```suggestion
    func test_sendReceipt_when_order_update_fails_then_send_receipt_fails() {
```